lib/datadog/profiling/component.rb

```diff
@@ -111,19 +111,30 @@ def self.build_profiler_component(settings:, agent_settings:, optional_tracer:)
       end
 
       private_class_method def self.enable_gc_profiling?(settings)
-        # See comments on the setting definition for more context on why it exists.
-        if settings.profiling.advanced.force_enable_gc_profiling
-          if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3')
-            Datadog.logger.debug(
-              'Profiling time/resources spent in Garbage Collection force enabled. Do not use Ractors in combination ' \
-              'with this option as profiles will be incomplete.'
-            )
-          end
+        return false unless settings.profiling.advanced.gc_enabled
 
-          true
-        else
-          false
+        # SEVERE - Only with Ractors
+        # On Ruby versions 3.0 (all), 3.1.0 to 3.1.3, and 3.2.0 to 3.2.2 gc profiling can trigger a VM bug
+        # that causes a segmentation fault during garbage collection of Ractors
+        # (https://bugs.ruby-lang.org/issues/18464). We don't allow enabling gc profiling on such Rubies.
+        # This bug is fixed on Ruby versions 3.1.4, 3.2.3 and 3.3.0.
+        if RUBY_VERSION.start_with?('3.0.') ||
+            (RUBY_VERSION.start_with?('3.1.') && RUBY_VERSION < '3.1.4') ||
+            (RUBY_VERSION.start_with?('3.2.') && RUBY_VERSION < '3.2.3')
+          Datadog.logger.warn(
+            "Current Ruby version (#{RUBY_VERSION}) has a VM bug where enabling GC profiling would cause "\
+            'crashes (https://bugs.ruby-lang.org/issues/18464). GC profiling has been disabled.'
+          )
+          return false
```

Comment:
Isn't this too aggressive since it only manifests when using Ractors (which bring other problems in those versions as well?).

Allocation suffers from the same issue and does not enforce false for this bug. We should either enforce there as well or we could have a separate flag governing disablement of both features related to ractor bugs?

Reply:
I'll concede it's a bit aggressive, and I shared my reasoning in the [Additional notes](https://github.com/DataDog/dd-trace-rb/pull/3558#:~:text=the%20timeline%20view.-,Additional%20Notes%3A,-Regarding%20the%20impact) for the commit.

TBH I think this feature is less important than allocation/heap, and so while we could put them on the same bucket, I... somewhat think the others deserve the extra attention, while this one doesn't quite?

Reply:
Fair fair 😄 And as you mention, we can revisit this at any point in the future anyway.